src/node/nodestate.h

```diff
@@ -408,12 +420,31 @@ namespace ccf
 #endif
       join_rpc.params.quote = quote;
 
-      auto join_req =
-        jsonrpc::pack(nlohmann::json(join_rpc), jsonrpc::Pack::MsgPack);
+      LOG_DEBUG_FMT(
+        "Sending join request to {}:{}",
+        args.config.joining.target_host,
+        args.config.joining.target_port);
+
+      join_client->send(jsonrpc::pack(join_rpc, jsonrpc::Pack::MsgPack));
+    }
+
+    void join(const Join::In& args)
+    {
+      std::lock_guard<SpinLock> guard(lock);
+      sm.expect(State::pending);
 
-      LOG_INFO_FMT("Sending join request");
+      initiate_join(args);
```

Comment:
The logic is: 
- When a joining node starts up, it first initiates a join RPC. 
- Straight after that, it creates a timer that ticks every second and that will initiate a new join if the node has not yet joined (i.e. still in state "pending").
- When the node has joined, the callback returns false and the timer stops ticking.

Reply:
Do we have a mermaid of this somewhere in the doc? If not, I think we possibly ought to.

Reply:
I think this should not be a concern for users/operators. The join timer is fast enough that the new node should join seamlessly after the proposal has completed.